### PR TITLE
[SPARK-40966][PS][TEST] Fix read_parquet test in pandas 1.5.1

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to enable the test_parquet_read_with_pandas_metadata with pandas 1.5.1.

### Why are the changes needed?

Because the bug introduced in 1.5.1 is not fixed yet, so we also handle the test with pandas 1.5.1

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

The existing CI should pass

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/pandas/tests/test_dataframe_spark_io.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def test_parquet_read_with_pandas_metadata(self):
             expected3 = expected2.set_index("index", append=True)
             # There is a bug in `to_parquet` from pandas 1.5.0 when writing MultiIndex.
             # See https://github.com/pandas-dev/pandas/issues/48848 for the reported issue.
-            if LooseVersion(pd.__version__) == LooseVersion("1.5.0"):
+            if LooseVersion(pd.__version__) in (LooseVersion("1.5.0"), LooseVersion("1.5.1")):
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/tests/test_dataframe_spark_io.py:102]: If they look not going to fix this soon, let's just do the comparison with `< LooseVersion("1.5.0")`

Comment (itholic) [reply python/pyspark/pandas/tests/test_dataframe_spark_io.py:102]: I'm not sure they it's fixed soon but seems trying to fix it ASAP, so I just wanna check explicitly for every version for now

Comment (itholic) [reply python/pyspark/pandas/tests/test_dataframe_spark_io.py:102]: If it wouldn't fixed in 1.5.2, let me check is as `< LooseVersion("1.5.0")` then ?

Comment (HyukjinKwon) [reply python/pyspark/pandas/tests/test_dataframe_spark_io.py:102]: Let's just fix it `< LooseVersion("1.5.0")` here.

Comment (itholic) [reply python/pyspark/pandas/tests/test_dataframe_spark_io.py]: Sure! Just addressed it

**Comment by itholic** [comment]
Applied this change to https://github.com/apache/spark/pull/38420. Closing

**Pull request closed without merging**